### PR TITLE
dapico-tools 0.1.4 (new formula)

### PR DESCRIPTION
**Pull request by nanoscopic**
## Summary
This PR adds dapico-tools, a small third-party utility providing
dapico-load and dapico-reboot for RP2040-based boards.

dapico-reboot reboots a connected device into BOOTSEL or normal
run mode. dapico-load loads an ELF image directly over USB.

The tools are macOS-only and use native IOKit USB APIs.
They are intentionally minimal and independent of the official
Raspberry Pi SDK and picotool.

## Testing
- Ran locally via a homebrew tap. ( on arm )
- Tested functionality on both arm and x86 with a pi pico board.
- Checked forumla with brew audit --strict
- Ran brew test ( which are just tests help runs as it needs a pi pico board )

### CODE REVIEW

File: Formula/d/dapico-tools.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class DapicoTools < Formula
+  desc "macOS USB tools for RP2040-based boards"
+  homepage "https://github.com/nanoscopic/dapico-tools"
+  url "https://github.com/nanoscopic/dapico-tools/archive/refs/tags/0.1.4.tar.gz"
+  sha256 "b39fa950f4080d5702ab00bc77ccc3c3ce057dc1b8963a1e3496607f0faccb29"
+  license "BSD-3-Clause"
```
Comment (chenrui333) [inline Formula/d/dapico-tools.rb:6]:
```suggestion
  license "BSD-3-Clause"

```

File: Formula/d/dapico-tools.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    system "#{bin}/dapico-reboot", "--help"
+    system "#{bin}/dapico-load", "--help"
```
Comment (chenrui333) [inline Formula/d/dapico-tools.rb:18]: need some better test

Comment (nanoscopic) [reply Formula/d/dapico-tools.rb:18]: And how am I supposed to add a test for a tool that communicates with a specific hardware device that none of the testing systems have?

Comment (nanoscopic) [reply Formula/d/dapico-tools.rb:18]: What is explicitly acceptable for hardware tools

For hardware-dependent tools, Homebrew accepts tests that:

run --help

run --version

run a command that fails cleanly with a deterministic error

confirm usage text

confirm exit codes

confirm no crash / no hang

There are many accepted formulae that do exactly this.

Examples (conceptually):

avrdude

openocd

dfu-util

bossac

picotool itself

None of these actually flash hardware in CI.

File: Formula/d/dapico-tools.rb (hunk 1 above)
Comment (nanoscopic) [reply Formula/d/dapico-tools.rb:6]: You want me to add a blank line after license line? Is that accurate?

File: Formula/d/dapico-tools.rb (hunk 2 above)
Comment (nanoscopic) [reply Formula/d/dapico-tools.rb:18]: Look at the test for picotool, which this is based on.
https://github.com/Homebrew/homebrew-core/blob/a1e6adad40f2432368c4c31a3bdb308b40a6e77d/Formula/p/picotool.rb

The only feature they test? Pulling info from a file. They don't test -any- of the other functionality such as load or reboot, which are the only features this tooling does.

That was accepted.

File: Formula/d/dapico-tools.rb (hunk 1 above)
Comment (nanoscopic) [reply Formula/d/dapico-tools.rb:6]: I see. I looked at the failing test and the output and have removed the spaces on the blank line.

File: Formula/d/dapico-tools.rb (hunk 2 above)
Comment (SMillerDev) [reply Formula/d/dapico-tools.rb:18]: Yes, because as documented in https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula we need something other than `--help` or `--version`. Those rarely exercise normal code paths, so do not give a good indication if something was compiled successfully. A test that fails on a common code path is therefore a much better test.

File: Formula/d/dapico-tools.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/d/dapico-tools.rb:6]: Yes please

File: Formula/d/dapico-tools.rb (hunk 2 above)
Comment (nanoscopic) [reply Formula/d/dapico-tools.rb:18]: That doesn't make any sense and accomplishes nothing but I have added a dryrun mode to load to comply with this. I am updating the repo and formula. Probably easiest to create a new PR at this point as trying to update all that some other way would be rather complex.

**Comment by nanoscopic** [comment]
Here is the new PR for anyone following along: https://github.com/Homebrew/homebrew-core/pull/263862

This new update uses a test.elf file and a dryrun mode to show what it would have done to load and run that elf had a pi pico been connected.

File: Formula/d/dapico-tools.rb (hunk 2 above)
Comment (nanoscopic) [reply Formula/d/dapico-tools.rb:18]: Fixed in new PR: https://github.com/Homebrew/homebrew-core/pull/263862
